### DISCUSSION TITLE
I still running keycloak 16 on production with no luck to upgrade, I can not find repository for keycloak 16

### DISCUSSION THREAD

**Discussion by elkashefehab** [keycloak/keycloak#28396] [Adapter deprecation]
I still running keycloak 16 on production with no luck to upgrade, I can not find repository for keycloak 16 to be able to build my image, I know its deprecated but I need the code to build it myself any help

**Comment by ThoreKr** [comment]
Have you looked through the tags?

https://github.com/keycloak/keycloak/tree/16.0.0

**Reply by elkashefehab** [reply]
found it thank you

**Reply by codespearhead** [reply]
@elkashefehab can you [mark it as answered](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) then?
